cristian-bars/challenge/precurse/Pasapalabra/pasapalabra.js

```diff
@@ -0,0 +1,147 @@
+    { letter: "v", answer: "vikingos", status: 0, question: "CON LA V. Nombre dado a los miembros de los pueblos nórdicos originarios de Escandinavia, famosos por sus incursiones y pillajes en Europa"},
+    { letter: "w", answer: "sandwich", status: 0, question: "CONTIENE LA W. Emparedado hecho con dos rebanadas de pan entre las cuales se coloca jamón y queso"},
+    { letter: "x", answer: "botox", status: 0, question: "CONTIENE LA X. Toxina bacteriana utilizada en cirujía estética"},
+    { letter: "y", answer: "peyote", status: 0, question: "CONTIENE LA Y. Pequeño cáctus conocido por sus alcaloides psicoactivos utilizado de forma ritual y medicinal  por indígenas americanos"},
+    { letter: "z", answer: "zen", status: 0, question: "CON LA Z. Escuela de budismo que busca la experiencia de la sabiduría más allá del discurso racional"}
+];
+
+function runGame(){
+    rigthWords = 0;
+    wrongWords = 0;
+    pendent = 0;
+    
+    console.log("Bienvenido al juego del Pasabalabra!!");
+    
+    getName();
+    runQuestions();
+    //Juego finalizado
+    console.log("Aquí finaliza el juego, el jugador " + yourName + " ha terminado la partida con un total de " + rigthWords + " palabras acertadas y un total de " + wrongWords + " palabras incorrectas!!");
+    
+    //Añadir jugador con sus resultados al array de jugadores
+    var obj = {};
+    obj["name"] = yourName;
+    obj["rigth"] = rigthWords;
+    obj["wrong"] = wrongWords;
+    
+    players.push(obj);
+    
+    //Preguntamos si quieren jugar otra partida
+    let playAgain = confirm("Quieres jugar otra partida??");
+        if (playAgain === true) {
+            for(let i = 0; i<questions.length;i++){
+                questions[i].status = 0;
+            }
+            runGame();
+        } else {
+            console.log("final");
+            sortParticipants(players);
+    
+            for(let i=0;i<players.length;i++){
+                console.log((i+1) + " " + players[i].name + " con " + players[i].rigth + " palabras acertadas y " + players[i].wrong + " palabras incorrectas");
+            }
+        } 
+    
+    
+}
+
+//Función para introducir el nombre del participante
+function getName(){
+    yourName = prompt("Por favor, indicanos tu nombre");
+    if (yourName === "" || yourName === null){
+        console.log("Tienes que poner un nombre!!");
+        getName();
+    }
+}
+
+//Función para ordenar los participantes por las palabras acertadas
+function sortParticipants(players){
+    players.sort(function (a, b) {
+      if (a.rigth > b.rigth) {
+        return -1;
+      }
+      if (a.rigth < b.rigth) {
+        return 1;
+      }
+      return 0;
+    });
+    
+}
+
+//Función para ir lanzando las preguntas
+//status == 0 pregunta en primer estado
+//status == 1 pregunta acertada
+//status == 2 pregunta incorrecta
+//status == 3 pasapalabra
+function runQuestions(){
+    for(let i = 0; i<questions.length;i++){
```

Comment:
For too simple for loops such as this one, it is better to replace with a for..in or a for..of loop.